### PR TITLE
[sc-522] Update hvm-lang to make it compatible with hvm-core n-ary nodes.

### PR DESCRIPTION
**Pull request by FranchuFranchu**

### CODE REVIEW

File: tests/snapshots/readback_lnet__invalid_mat_mat.hvm.snap [hunk 1]
```diff
@@ -2,8 +2,5 @@
 source: tests/golden_tests.rs
 input_file: tests/golden_tests/readback_lnet/invalid_mat_mat.hvm
 ---
-Warnings:
-During readback:
-  Invalid Numeric Match. (4 occurrences)
-
-λa match match <Invalid> { 0: 3; 1+*: 4 } { 0: <Invalid>; 1+: <Invalid> }
+Errors:
+invalid legacy match
```
Comment (developedby) [inline tests/snapshots/readback_lnet__invalid_mat_mat.hvm.snap:6]: Where is this error message comming from?

Comment (tjjfvi) [reply tests/snapshots/readback_lnet__invalid_mat_mat.hvm.snap:6]: https://github.com/HigherOrderCO/hvm-core/blob/main/src/ast.rs#L325

Comment (tjjfvi) [reply tests/snapshots/readback_lnet__invalid_mat_mat.hvm.snap:6]: `?<a b>` is no longer valid syntax: https://github.com/HigherOrderCO/hvm-lang/blob/be2dcf4203237f6ff2d22aa7c22a720fc07541f9/tests/golden_tests/readback_lnet/invalid_mat_mat.hvm#L3C17-L3C23

**Review by developedby — APPROVED** [review]

File: tests/snapshots/readback_lnet__invalid_mat_mat.hvm.snap (hunk 1 above)
Comment (tjjfvi) [reply tests/snapshots/readback_lnet__invalid_mat_mat.hvm.snap:6]: correct; that test case no longer makes sense and can/should be removed

**Pull request merged**